File: tests/fixtures/polyglot_baseline/python_service/models/organization.py
"""
Organization models for request/response validation.

Defines Pydantic models for organization data with validation.
"""

from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field

from .permissions import RoleEnum


class OrgCreate(BaseModel):
    """Request model for organization creation."""

    name: str = Field(..., min_length=1, max_length=100, description="Organization name")
    description: Optional[str] = Field(
        None, max_length=500, description="Organization description"
    )


class OrgUpdate(BaseModel):
    """Request model for organization updates."""

    name: Optional[str] = Field(None, min_length=1, max_length=100)
    description: Optional[str] = Field(None, max_length=500)


class OrgMemberRole(BaseModel):
    """Request model for organization member role assignment."""

    user_id: int = Field(..., description="User ID")
    role: RoleEnum = Field(..., description="Role in organization")


class OrgMember(BaseModel):
    """Organization member information."""

    user_id: int = Field(..., description="User ID")
    username: str = Field(..., description="Username")
    email: str = Field(..., description="Email")
    role: RoleEnum = Field(..., description="Role in organization")
    joined_at: datetime = Field(..., description="Joined timestamp")

    class Config:
        """Pydantic config."""
        from_attributes = True


class OrgResponse(BaseModel):
    """Response model for organization data."""

    id: int = Field(..., description="Organization ID")
    name: str = Field(..., description="Organization name")
    description: Optional[str] = Field(None, description="Organization description")
    owner_id: int = Field(..., description="Owner user ID")
    member_count: int = Field(default=1, description="Number of members")
    created_at: datetime = Field(..., description="Creation timestamp")
    updated_at: datetime = Field(..., description="Last update timestamp")

    class Config:
        """Pydantic config."""
        from_attributes = True


class OrgDetailResponse(OrgResponse):
    """Detailed organization response with members."""

    members: list[OrgMember] = Field(default_factory=list, description="Organization members")

File: tests/fixtures/polyglot_baseline/go_backend/models/organization.go
package models

import "time"

type Organization struct {
	ID          int64     `json:"id"`
	Name        string    `json:"name"`
	Description string    `json:"description"`
	OwnerID     int64     `json:"owner_id"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
	IsActive    bool      `json:"is_active"`
}

type OrgMember struct {
	ID     int64  `json:"id"`
	OrgID  int64  `json:"org_id"`
	UserID int64  `json:"user_id"`
	Role   string `json:"role"` // admin, member, viewer
	AddedAt time.Time `json:"added_at"`
}

func NewOrganization(name, description string, ownerID int64) *Organization {
	return &Organization{
		Name:        name,
		Description: description,
		OwnerID:     ownerID,
		CreatedAt:   time.Now(),
		UpdatedAt:   time.Now(),
		IsActive:    true,
	}
}
